lib/lev/background_job.rb

```diff
@@ -113,22 +119,25 @@ def self.store
       Lev.configuration.job_store
     end
 
+    def self.fetch_and_parse(job_key)
+      fetched = store.fetch(job_key)
+      return nil if fetched.nil?
+      JSON.parse(fetched).stringify_keys!
+    end
+
     def self.job_ids
       store.fetch(job_key('lev_job_ids')) || []
     end
 
     def stored
-      if found = self.class.store.fetch(job_key)
-        JSON.parse(found)
-      else
-        {}
-      end
+      self.class.fetch_and_parse(job_key) || {}
     end
 
     def track_job_id
       ids = self.class.job_ids
+      return if ids.include?(@id)
       ids << @id
-      self.class.store.write(self.class.job_key('lev_job_ids'), ids.uniq)
+      self.class.store.write(self.class.job_key('lev_job_ids'), ids)
```

Comment:
The before and after here may look equivalent, but the prior approach gave us different string objects from call to call even if the ID already existed.  This caused the `all` implementation to return duplicate `BackgroundJob`s.